src/dune_engine/scheduler.ml

```diff
@@ -643,8 +676,101 @@ module Handler = struct
   type t = Config.t -> Event.t -> unit
 end
 
+module Timer : sig
```

Comment:
Should we rename `Timer.t` to `Timer_pool.t` or something else (e.g. `Alarm_clock`) that isn't so easy to confuse with a singular scheduled event?

For the `sleeper` concept, we have precedent `Timing_wheel.Alarm.t` and `Async.Clock.Event.t`. "Event" is way overloaded, but "alarm" sounds unused and I like it better than "sleeper".

Reply:
I went with Alaram_clock/alarm